File: API/WebApplication1/Models/Nedetid.cs
namespace WebApplication1.Models
{
    public enum Hverdag
    {
        Monday,
        Tuesday,
        Wednesday,
        Thursday,
        Friday,
        Saturday,
        Sunday
    }

    public class Nedetid : Common
    {
        public required Hverdag Dag { get; set; }
        public required TimeOnly Tidspunkt { get; set; }
        public required bool Gentagende { get; set; }
        public required DateTime Start { get; set; }
        public required DateTime End { get; set; }
        public required string MaskineId { get; set; }
        public Maskine? Maskine { get; set; }
    }
    // UDEN ID
    public class NedetidDTO
    {
        public required Hverdag Dag { get; set; }
        public required TimeOnly Tidspunkt { get; set; }
        public required bool Gentagende { get; set; }
        public required DateTime Start { get; set; }
        public required DateTime End { get; set; }
        public required string MaskineId { get; set; }
    }
}

File: Flutter/assignly/lib/classes/objects/nedetid.dart
// ignore_for_file: constant_identifier_names
import 'dart:convert';

enum Hverdag {
  Monday,
  Tuesday,
  Wednesday,
  Thursday,
  Friday,
  Saturday,
  Sunday
}

class Nedetid {
  String? id;
  Hverdag? dag;
  // TIMEONLY FORMAT HH:MM
  String? tidspunkt;
  bool? gentagende;
  DateTime? start;
  DateTime? end;

  static Nedetid getNedetidFromJson(String response) {
    try {
      final data = jsonDecode(response);
      final results = data;

    } catch (_) {
    }
    return Nedetid();
  }
}
